### PR TITLE
Exception handling for set device locale feature

### PR DESCRIPTION
**Pull request by artem888**
## Proposed Changes

This PR adds better exception handling for setting a device locale feature:
- improves cli error messages and specifications about `--device-locale` in case of exceptions
- adds a list of supported languages and countries by Maestro
- in case of Android we can catch some exceptions happened in the driver via `resultCode` and `resultData` received from BroadcastReceiver

## Testing
- [x] local test
- [ ] staging test

### CODE REVIEW

File: maestro-cli/src/main/java/maestro/cli/command/StartDeviceCommand.kt [hunk 1]
```diff
@@ -87,13 +87,99 @@ class StartDeviceCommand : Callable<Int> {
...
+
+    private fun validateLocale(language: String, country: String) {
+        if (!SUPPORTED_LANGUAGES.map { it.first }.contains(language)) {
+            val languages = SUPPORTED_LANGUAGES.joinToString("\n")
+            throw CliError("$language language is currently not supported by Maestro, please check that it is a valid ISO-639-1 code. Here is a full list of supported languages:\n" +
+                    "\n" +
+                    languages
+            )
+        }
+        if (!SUPPORTED_COUNTRIES.map { it.first }.contains(country)) {
```
Comment (amanjeetsingh150) [inline maestro-cli/src/main/java/maestro/cli/command/StartDeviceCommand.kt:111]: How are we getting supported countries and languages? I see this hardcoded.

File: maestro-cli/src/main/java/maestro/cli/device/DeviceService.kt [hunk 2]
```diff
@@ -76,10 +76,13 @@ object DeviceService {
                     PrintUtils.message("Setting the device locale to ${device.language}_${device.country}...")
                     val driver = AndroidDriver(dadb)
                     driver.open()
-                    driver.setDeviceLocale(
+                    val result = driver.setDeviceLocale(
                         country = device.country,
                         language = device.language
                     )
+                    if (result != -1) {
+                        throw IllegalStateException("Android failed to update a device locale")
```
Comment (amanjeetsingh150) [inline maestro-cli/src/main/java/maestro/cli/device/DeviceService.kt:84]: This is not debuggable really I think we should have failure reasons here and throw proper errors with message

Comment (amanjeetsingh150) [reply maestro-cli/src/main/java/maestro/cli/device/DeviceService.kt:84]: Also how the failure in setting locale works on iOS?

File: maestro-cli/src/main/java/maestro/cli/command/StartDeviceCommand.kt [hunk 3]
```diff
@@ -87,13 +87,99 @@ class StartDeviceCommand : Callable<Int> {
...
+            throw CliError("$country country is currently not supported by Maestro, please check that it is a valid ISO-3166-1 code. Here is a full list of supported countries:\n" +
+                    "\n" +
+                    countries
+            )
+        }
+    }
+
+    companion object {
+        // ISO-639-1
+        private val SUPPORTED_LANGUAGES = listOf(
```
Comment (axelniklasson) [inline maestro-cli/src/main/java/maestro/cli/command/StartDeviceCommand.kt:122]: Did we verify that this list is the same for both Android and iOS?

File: maestro-client/src/main/java/maestro/drivers/AndroidDriver.kt [hunk 4]
```diff
@@ -568,9 +568,16 @@ class AndroidDriver(
         LOGGER.info("[Done] Adding media files")
     }
 
-    fun setDeviceLocale(country: String, language: String) {
+    fun setDeviceLocale(country: String, language: String): Int {
         dadb.shell("pm grant dev.mobile.maestro android.permission.CHANGE_CONFIGURATION")
-        dadb.shell("am broadcast -a dev.mobile.maestro.locale -n dev.mobile.maestro/.receivers.LocaleSettingReceiver --es lang $language --es country $country")
+        val response = dadb.shell("am broadcast -a dev.mobile.maestro.locale -n dev.mobile.maestro/.receivers.LocaleSettingReceiver --es lang $language --es country $country")
```
Comment (axelniklasson) [inline maestro-client/src/main/java/maestro/drivers/AndroidDriver.kt:573]: How will we surface any error thrown by this call?

File: maestro-cli/src/main/java/maestro/cli/device/DeviceService.kt (hunk 2 above)
Comment (axelniklasson) [reply maestro-cli/src/main/java/maestro/cli/device/DeviceService.kt:84]: +1 on what Amanjeet said, we should have as descriptive error messages as possible and avoid these catch all messages if we can. Do we know what can go wrong here? If yes, can we categorize it? If no, let's maybe display it to the user?

Comment (artem888) [reply maestro-cli/src/main/java/maestro/cli/device/DeviceService.kt]: added more helpful failure reasons for Android and made exception message more descriptive

Comment (artem888) [reply maestro-cli/src/main/java/maestro/cli/device/DeviceService.kt]: > Also how the failure in setting locale works on iOS?

There is sadly no way to verify that `xcrun simctl spawn defaults write` will always be successful. I think good documentation and exception messages are enough here for the users to be able to debug the reason

File: maestro-cli/src/main/java/maestro/cli/command/StartDeviceCommand.kt (hunk 1 above)
Comment (artem888) [reply maestro-cli/src/main/java/maestro/cli/command/StartDeviceCommand.kt]: decided to remove hardcoded values here, explained the reasoning offline

File: maestro-cli/src/main/java/maestro/cli/command/StartDeviceCommand.kt (hunk 3 above)
Comment (artem888) [reply maestro-cli/src/main/java/maestro/cli/command/StartDeviceCommand.kt]: answered offline

File: maestro-client/src/main/java/maestro/drivers/AndroidDriver.kt (hunk 4 above)
Comment (artem888) [reply maestro-client/src/main/java/maestro/drivers/AndroidDriver.kt:573]: this call is always successful , exceptions happen on android app level, handled them by propagating certain error codes and data messages for now

**Review by amanjeetsingh150 — APPROVED** [review]

**Pull request merged**